components/browser/menu/src/main/java/mozilla/components/browser/menu/item/BrowserMenuHighlightableItem.kt

```diff
@@ -94,40 +102,90 @@ class BrowserMenuHighlightableItem(
+                        endImageView.visibility = View.VISIBLE
+                    }
                 }
-            }
-            with(endImageView) {
-                if (highlight.endImageResource != NO_ID) {
-                    setImageResource(highlight.endImageResource)
-                    visibility = View.VISIBLE
+                is BrowserMenuHighlight.LowPriority -> {
+                    val drawable = createNotificationDrawable(view.context, highlight.notificationTint)
+                    textView.text = highlight.label ?: label
+                    startImageView.setImageDrawable(drawable)
+                    view.background = null
+                }
+                is BrowserMenuHighlight.ClassicHighlight -> {
+                    textView.text = label
+                    view.setBackgroundResource(highlight.backgroundResource)
+                    if (highlight.startImageResource != NO_ID) {
+                        startImageView.setImageResource(highlight.startImageResource)
+                    }
+                    if (highlight.endImageResource != NO_ID) {
+                        endImageView.setImageResource(highlight.endImageResource)
+                        endImageView.visibility = View.VISIBLE
+                    }
                 }
             }
         } else {
-            val selectableItemBackground =
-                view.context.theme.resolveAttribute(android.R.attr.selectableItemBackground)
+            textView.text = label
+            view.background = null
+            startImageView.setImageDrawable(startDrawable)
+            endImageView.setImageDrawable(null)
+            endImageView.visibility = View.GONE
+        }
+    }
+
+    private fun createStartDrawable(context: Context): Drawable? {
+        if (startImageDrawable == null && startImageResource != NO_ID) {
+            startImageDrawable = AppCompatResources.getDrawable(context, startImageResource)
+            if (iconTintColorResource != NO_ID) {
+                val tint = ContextCompat.getColor(context, iconTintColorResource)
+                startImageDrawable = startImageDrawable?.mutate()?.apply { setTint(tint) }
+            }
+        }
+
+        return startImageDrawable
+    }
 
-            view.setBackgroundResource(selectableItemBackground)
-            with(endImageView) {
-                setImageResource(0)
-                visibility = View.GONE
+    private fun createNotificationDrawable(context: Context, @ColorInt tint: Int): Drawable {
+        if (notificationDrawable == null) {
+            val dot = AppCompatResources.getDrawable(
+                context,
+                R.drawable.mozac_browser_menu_notification_icon
+            )?.mutate()?.apply {
+                setTint(tint)
             }
+            notificationDrawable = LayerDrawable(arrayOf(startImageDrawable, dot))
         }
+
+        return notificationDrawable!!
```

Comment:
nit: I think we could avoid the ‼️  by using an auxiliary function.

```kotlin
    private fun createNotificationDrawable(context: Context, @ColorInt tint: Int): Drawable {
        return notificationDrawable ?: newDrawable(context, tint)
    }

    private fun newDrawable(context: Context, @ColorInt tint: Int): Drawable {
        val dot = AppCompatResources.getDrawable(
                context,
                R.drawable.mozac_browser_menu_notification_icon
        )?.mutate()?.apply {
            setTint(tint)
        }
        return LayerDrawable(arrayOf(startImageDrawable, dot))
    }
```

Reply:
I'll rewrite this with let or apply, because I'd prefer to avoid an extra function for this extra function 😄

Reply:
Actually it's fine to return nullable here, `setDrawable` doesn't care.